lib/src/wallet/transactions.rs

```diff
@@ -332,22 +394,38 @@ impl WalletTxns {
     pub fn mark_txid_nf_spent(
         &mut self,
         txid: TxId,
-        nullifier: &SaplingNullifier,
+        nullifier: &WalletNullifier,
         spent_txid: &TxId,
         spent_at_height: BlockHeight,
     ) -> u64 {
-        let mut note_data = self
-            .current
-            .get_mut(&txid)
-            .unwrap()
-            .sapling_notes
-            .iter_mut()
-            .find(|n| n.nullifier == *nullifier)
-            .unwrap();
-
-        note_data.spent = Some((spent_txid.clone(), spent_at_height.into()));
-        note_data.unconfirmed_spent = None;
-        note_data.note.value
+        match nullifier {
+            WalletNullifier::Sapling(nf) => {
+                let mut note_data = self
+                    .current
+                    .get_mut(&txid)
+                    .unwrap()
+                    .sapling_notes
+                    .iter_mut()
+                    .find(|n| n.nullifier == *nf)
+                    .unwrap();
+                note_data.spent = Some((spent_txid.clone(), spent_at_height.into()));
```

Comment:
`note_data.spent` should be a three variant enum

```Rust
enum Spent {  Confirmed, Unconfirmed, None }
```
`note_data.unspent` should be removed.

Reply:
Non-blocking, tracking with: #54